pkgs/development/libraries/gpgme/default.nix

```diff
@@ -30,6 +30,9 @@ stdenv.mkDerivation rec {
   NIX_CFLAGS_COMPILE =
     with stdenv; lib.optional (system == "i686-linux") "-D_FILE_OFFSET_BITS=64";
 
+  AM_CXXFLAGS =
+    with stdenv; lib.optional (isDarwin) "-D_POSIX_C_SOURCE=200809L";
```

Comment:
`with` usage follows the pattern of the previous line; as for being conditional, I’m not sure: on one hand, it looks like `strdup` is only specified to be defined if one of the several macros is defined; on the other hand—it used to work, so why fix what’s not broken on other platforms?